### DISCUSSION TITLE
Using Number.toString()

### DISCUSSION THREAD

**Discussion by shark-in-a-hat** [brython-dev/brython#2218] [Q&A] [answered]
Javascript Number (besides obviously being the Devil's work) has a nifty .toString() method that lets you convert a base-10 integer to any other base. As far as I know, there's nothing like that (build in) in Python, and you have to write something custom that may be slow, recursive and/or import the entire universe. 

In Brython you can import the javascript Number, but I've no idea how to use it:

```
Brython 3.11.2 on Netscape 5.0 (Windows)
Type "help", "copyright", "credits" or "license" for more information.
>>> from javascript import Number as js_num
>>> x = js_num(128)
>>> x
128
>>> x.toString(32)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'str' object is not callable

```

Writing a helper function in javascript isn't that hard, but I'd like to avoid the need

```html
<script type="text/javascript">
        function to_base_n(value, n){
            return Number(value).toString(n)
        }
</script> 
```
Any tips or ideas how this can be achieved using pure Brython?

**Comment by BobSquarePants** [comment]
Hi @shark-in-a-hat , I guess you will have your answer there --> https://python-forum.io/thread-39151-post-166130.html#pid166130

**Reply by shark-in-a-hat** [reply]
I'm more interested in how to get my hands on the .toString method of the javascript Number object from brython than how to convert numbers to a different base (even though that was the original goal). 
Thanks for the reply anyway.

**Comment by PierreQuentel** [comment, marked as answer]
Hello @shark-in-a-hat ,

The error message is misleading so I keep the issue open, but you can use `.toString()` if you create the number with the Brython equivalent of Javascript's `new Number(128)`:

```
Brython 3.11.2 on Netscape 5.0 (Windows)
Type "help", "copyright", "credits" or "license" for more information.
>>> from javascript import Number as js_num
>>> x = js_num.new(128)
>>> x
<Javascript object: [object Number]>
>>> x.toString(32)
'40'
>>> 
```

**Reply by shark-in-a-hat** [reply]
That's something I didn't think of trying. Works good, solves my issue, thanks.
